src/libs/ReportUtils.ts

```diff
@@ -892,16 +901,24 @@ function getChatType(report: OnyxInputOrEntry<Report> | Participant): ValueOf<ty
 /**
  * Get the report or draft report given a reportID
  */
-function getReportOrDraftReport(reportID: string | undefined): OnyxEntry<Report> {
-    return allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] ?? allReportsDraft?.[`${ONYXKEYS.COLLECTION.REPORT_DRAFT}${reportID}`];
+function getReportOrDraftReport(reportID: string | undefined, reports?: SearchReport[]): OnyxEntry<Report> | SearchReport {
+    if (reports) {
+        return reports?.find((report) => report.reportID === reportID);
+    }
+
+    if (allReports) {
+        return allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`];
+    }
```

Comment:
I believe you might've changed the logic a bit by mistake.

Previously the flow went like this:
 - IF `allReports` not empty, then try to grab a reportID
 - then if report was undefined/null fallback to searching in `allReportsDraft`

But now as long as `allReports` is non-empty (which it will always be I think) we just return. So if report is missing we will return `undefined` and not look in `allReportsDraft` which we should.